feat: improve weekly tests ci utilisation

DavePearce opened the pull request.

DavePearce committed:
tweak max heap limit for tests

This tweaks the max heap limit to ensure the right value is given when
running on the CI server.

DavePearce committed:
fix bottleneck with SignedOperationsExtensiveTest

There was a performance bottleneck with the test
SignedOperationsExtensiveTest which was using a MethodOrderer and a
Lifecyle.CLASS attribute.  Both of these meant that the default
parallelism mode of concurrent was being ignored.  That mean this one
test was significantly limiting the amount of parallelism on the CI
machine.

DavePearce committed:
Enable concurrent execution for specific tests

These tests make use of the `TestInstance.Lifecycle.PER_CLASS` attribute
which prohibit concurrent execution.  To get that, we have to add
another attribute `@Execution(ExecutionMode.CONCURRENT)`

DavePearce committed:
tweaking prc-calltests

lorenzogentile404 commented on arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/callTests/sixtyThreeSixtyFourths/SixtyThreeSixtyFourthsTests.java:
```diff
@@ -80,6 +82,7 @@
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@Execution(ExecutionMode.CONCURRENT)
 public class SixtyThreeSixtyFourthsTests {
```
`@TestInstance(TestInstance.Lifecycle.PER_CLASS)` seems to be not necessary anymore as the method sources may be static.

DavePearce replied:
Ok, because they no longer use a `static Random` field?  At least I can't see one.

DavePearce replied:
I will remove this then.

lorenzogentile404 commented on arithmetization/src/test/java/net/consensys/linea/zktracer/SignedOperationsExtensiveTest.java:
```diff
@@ -51,16 +49,18 @@ void signedComparisonsModDivTest(OpCode opCode, String a, String b) {
-  Stream<Arguments> signedComparisonsModDivTestSource() {
+  private static Stream<Arguments> signedComparisonsModDivTestSource() {
+    Random rng = new Random(123);
+
```
What is the advantage of declaring `rng` inside the method source here? Then we need to pass it every time to `randomBytes`, that seems to be not ideal. Is there an advantage from an efficiency perspective?

DavePearce replied:
Well, there are pros / cons.  This is just one way to solve the problem.  The other is to use `@Execution(ExecutionMode.CONCURRENT)` which I did in other cases.  We can do that here, though passing `rng` to `randomBytes` everytime is not a big deal from my perspective.  Personally, if it were me, I would remove as many annotations as possible and stick with the simplest approach available.  Here, we have removed `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` which, already, caused problems (where as for the other examples we have to add another annotation to compensate).  I did add in `@Accessors(fluent = true)` and `@ExtendWith(UnitTestWatcher.class)` to be consistent with other tests, though I am unsure whether they are strictly necessary.  If not, they should be removed across the board.

DavePearce committed:
remove `Lifecycle.PER_CLASS` from test

This removes this annotation from the test class `SixtyThreeSixtyFourthsTests`
 by reworking the sources of the parameterised tests to be static.

lorenzogentile404 reviewed (approved):
LGTM
